### PR TITLE
Added error abstraction to tzip16 and modified dependant package (tzi…

### PR DESCRIPTION
**Pull request by dsawali**
…p12)

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__
- Added better error abstraction for tzip16 view calls

### CODE REVIEW

File: packages/taquito-tzip16/src/viewKind/michelson-storage-view.ts [hunk 1]
```diff
@@ -169,7 +170,26 @@ export class MichelsonStorageView implements View {
             balance: '0'
         };
 
-        const result: any = await this.rpc.runCode(viewScript as any);
+        let result: any;
+        try {
+          result = await this.rpc.runCode(viewScript as any);
+        } catch(error: any) {
+          const parsedError = JSON.parse(error.body);
```
Comment (roxaneletourneau) [inline packages/taquito-tzip16/src/viewKind/michelson-storage-view.ts:177]: I am wondering if there is a risk of having an error body that is an invalid JSON, JSON.parse would result in an "Unexpected token.." error
I have a similar comment for the .forEach at line 179, I am wondering what will be the behavior if the parsed body is not an array.

I think that the `extractFailedWithDetails` function could be reused here. https://github.com/ecadlabs/taquito/blob/master/packages/taquito/src/contract/errors.ts#L66
If the body is valid JSON, and if it contains an array of errors, it extracts the "with" property of the last error which contains the information about the script execution failure.

File: packages/taquito-tzip16/src/viewKind/michelson-storage-view.ts [hunk 2]
```diff
@@ -169,7 +170,26 @@ export class MichelsonStorageView implements View {
...
+
+          parsedError.forEach((x: any) => {
+            if (x.id.includes('script_rejected')) {
+              throw new ViewSimulationError(`Failed to execute the contract view ${parsedError}`)
+            } 
+          });
+          throw new HttpResponseError(
```
Comment (roxaneletourneau) [inline packages/taquito-tzip16/src/viewKind/michelson-storage-view.ts:184]: consider replacing by `throw error`

**Review by roxaneletourneau — CHANGES_REQUESTED** [review]

File: packages/taquito-tzip16/src/viewKind/michelson-storage-view.ts (hunk 1 above)
Comment (dsawali) [reply packages/taquito-tzip16/src/viewKind/michelson-storage-view.ts:177]: that's a very good point. `extractFailedWithDetails` seem to have all the checks we need, so I will be using that. Will amend

File: packages/taquito-tzip16/src/viewKind/michelson-storage-view.ts (hunk 2 above)
Comment (dsawali) [reply packages/taquito-tzip16/src/viewKind/michelson-storage-view.ts:184]: I'm guessing we want that for the sake of generality?

File: packages/taquito-tzip16/src/viewKind/michelson-storage-view.ts [hunk 3]
```diff
@@ -169,7 +170,14 @@ export class MichelsonStorageView implements View {
             balance: '0'
         };
 
-        const result: any = await this.rpc.runCode(viewScript as any);
+        let result: any;
+        try {
+          result = await this.rpc.runCode(viewScript as any);
+        } catch(error: any) {
+          const failWith = validateAndExtractFailwith(error)
+          throw failWith? new ViewSimulationError(`The Simulation of the on-chain view failed with: ${JSON.stringify(failWith)}`, error): error;
```
[the remark this answers is not in the document]
Comment (dsawali) [reply packages/taquito-tzip16/src/viewKind/michelson-storage-view.ts:178]: Ah I thought that michelson views are considered on-chain. Thanks for clearing that up. Will fix wording

**Review by roxaneletourneau — APPROVED** [review]

**Pull request merged**